### PR TITLE
Parse Ruby comments as CommonMark

### PR DESCRIPTION
**Pull request by eoogbe**
Updates the formatting of Ruby comments to expect comments in the format 
of [CommonMark](https://commonmark.org/). Proto comments generally 
follow the CommonMark spec.

Replaces ad hoc comment transformations with a visitor for the Markdown 
elements. Adds tests for the RubyCommentReformatter.

Cleans up comment regex patterns.

Updates #722

### CODE REVIEW

**Comment by andreamlin** [comment]
seems good to me, but I completely lack contextual knowledge about commonmark

**Comment by andreamlin** [comment]
maybe the failures are DIRT related?

**Comment by eoogbe** [comment]
@landrito PTAL

**Comment by jbolinger** [comment]
@eoogbe what's the reason for this change? The baseline changes don't look too significant but I'm worried that this could break existing synth scripts for the Ruby libraries.

**Comment by eoogbe** [comment]
It ensures the Ruby comments consistently follow CommonMark, which the form of markdown proto comments use. The main reason for the change is to make the CommentReformatter more testable. See #722

File: src/main/java/com/google/api/codegen/util/ruby/RubyCommentReformatter.java [hunk 1]
```diff
@@ -14,106 +14,256 @@
...
-          .transform(LinkPattern.ABSOLUTE.toFormat("[$TITLE]($URL)"))
-          .scopedReplace(CommentPatterns.HEADLINE_PATTERN, "#", "=")
-          .build();
+
+  private static final Logger LOGGER = Logger.getLogger(RubyCommentReformatter.class.getName());
+
+  // Might as well create only one. Parser is thread-safe.
+  private static final Parser PARSER = Parser.builder().build();
+
+  private static String CLOUD_URL_PREFIX = "https://cloud.google.com";
```
Comment (jbolinger) [inline src/main/java/com/google/api/codegen/util/ruby/RubyCommentReformatter.java]: Is this duplicated now?

File: src/test/java/com/google/api/codegen/gapic/testdata/ruby/ruby_library.baseline [hunk 2]
```diff
@@ -1721,9 +1717,7 @@ module Google
     # method. In Python, a standard `datetime.datetime` object can be converted
     # to this format using [`strftime`](https://docs.python.org/2/library/time.html#time.strftime)
     # with the time format spec '%Y-%m-%dT%H:%M:%S.%fZ'. Likewise, in Java, one
-    # can use the Joda Time's [`ISODateTimeFormat.dateTime()`](
-    # http://www.joda.org/joda-time/apidocs/org/joda/time/format/ISODateTimeFormat.html#dateTime--
-    # ) to obtain a formatter capable of generating timestamps in this format.
+    # can use the Joda Time's [`ISODateTimeFormat.dateTime()`](http://www.joda.org/joda-time/apidocs/org/joda/time/format/ISODateTimeFormat.html#dateTime--) to obtain a formatter capable of generating timestamps in this format.
```
Comment (jbolinger) [inline src/test/java/com/google/api/codegen/gapic/testdata/ruby/ruby_library.baseline]: Why is this changing?

File: src/test/java/com/google/api/codegen/gapic/testdata/ruby/ruby_library.baseline [hunk 3]
```diff
@@ -1991,7 +1984,9 @@ module Google
         #
         #     * Here is a sentence about the first element of the list that continues
         #       into a second line.
+        #
```
Comment (jbolinger) [inline src/test/java/com/google/api/codegen/gapic/testdata/ruby/ruby_library.baseline]: Why are these newlines being added?

**Comment by eoogbe** [comment]
@jbolinger If it's the case that we don't want to fix the comment formatting, then we should probably close the issue.

**Pull request closed without merging**